weather-app+dinamic map/index.js

```diff
@@ -0,0 +1,117 @@
+L.control.layers({
+    "Streets": streets
+}).addTo(map); 
+
+function renderTbody() {
+    tbody.innerHTML = '' 
+    storage.forEach((value, key) => {
+        const [city, country, temperature] = value
+        const row = `<tr>
+            <td>${city}</td>
+            <td>${country}</td>
+            <td>${temperature}</td>
+            <td>${temperature}</td>
+        </tr>`
+        tbody.innerHTML = tbody.innerHTML + row
+    })
+}
+
+buttonMyWeather.addEventListener('click', function(){
+    
+        navigator.geolocation.getCurrentPosition(function(position) {
+            const latMyWeather = position.coords.latitude;
+            const lonMyWeather = position.coords.longitude;
+            console.log(latMyWeather, lonMyWeather);
+            fetch(`http://api.weatherstack.com/current?access_key=c3f9ea88c1f45ba3d7f06dbe3ac4fa49&query= ${latMyWeather}, ${lonMyWeather}`)    
+                .then(result => result.json())
+                .then(data => {
+                const {
+                    location: {name, country},
+                    current: {temperature}
+                } = data
+                storage.set(name, [name, country, temperature]);
+                renderTbody();
+                marker = L.marker([latMyWeather, lonMyWeather]).addTo(map);
+                marker.bindPopup(`My location: ${name}
+                My temperature: ${temperature}`).openPopup();
+                mapMarkers.push(marker);
+                })
+            })
+})
+
+buttonFind.addEventListener('click', function(){
+    console.log(input.value);
+    if (localStorage){ 
+        let keys = Object.keys(localStorage);
+        console.log(keys)
+        for (let key in keys){
+            console.log(keys[key])
+            fetch(`http://api.weatherstack.com/current?access_key=c3f9ea88c1f45ba3d7f06dbe3ac4fa49&query= ${keys[key]}`)
+            .then(result => result.json())
+            .then(data => {
+                const {
+                    location: {name, country, lat, lon},
+                    current: {temperature}
+                } = data
+                storage.set(name, [name, country, temperature]);
+                renderTbody();
+                marker = L.marker([lat, lon]).addTo(map);
+                marker.bindPopup(`${[lat, lon]}`).openPopup();
+                mapMarkers.push(marker);
+            })
+        }
+        
+    } 
+    if (!!localStorage){
+        fetch(`http://api.weatherstack.com/current?access_key=c3f9ea88c1f45ba3d7f06dbe3ac4fa49&query= ${input.value}`)
+            .then(result => result.json())
+            .then(data => {
+                const {
+                    location: {name, country, lat, lon},
+                    current: {temperature}
+                } = data
+                storage.set(name, [name, country, temperature]);
+                renderTbody();
+                arr.push(name);
+                console.log(arr)
+                localStorage.setItem(`${name}`, [arr])
```

Comment:
why you put all data in every name field ? You will get strange storage like: 

```
 localStorage 
{
'minsk': ['minsk'],
'brest': ['minsk', 'brest'],
'gomel': ['minsk', 'brest', 'gomel']
}
```

instead save all data in constant key like

 `localStorage.setItem('cities', arr)`